Bulk actions framework

mejackreed opened the pull request:
Closes #136 

This pull request lays the foundation for the future of Argo's bulk actions by adding a framework and implementing descriptive metadata download.

![screen shot 2016-02-08 at 5 21 52 pm](https://cloud.githubusercontent.com/assets/1656824/12905021/8b644cee-ce88-11e5-8867-83745c51f0e5.png)
![screen shot 2016-02-08 at 5 21 21 pm](https://cloud.githubusercontent.com/assets/1656824/12905022/8b646d3c-ce88-11e5-8855-570de7a0a201.png)
![screen shot 2016-02-08 at 5 21 38 pm](https://cloud.githubusercontent.com/assets/1656824/12905023/8b64b148-ce88-11e5-8a94-34d0e1e42703.png)

mejackreed commented on app/jobs/generic_job.rb:
```diff
@@ -0,0 +1,22 @@
+##
```
I think this is a miscue by Hound.

drh-stanford commented:
Re: schema. Since the `action_type` and `status` columns both seem to be enumerated types, it'd be nice to see somewhere what all the possible values are for them. Ideally with a check constraint on the schema.

mejackreed commented:
Thanks for the comment @drh-stanford . In 8db6f29 I've added the validation for `action_type` at the model level and added a test for this. Is that sufficient?

drh-stanford commented:
What about `status`? Or do you never query on status?

mejackreed commented:
Ah.. yeah.. we don't query by status but we do check if "Completed". Hmm.. I feel like that is ok with being left more free form as we aren't executing any user provided params here. Thoughts?

drh-stanford commented:
If it's meant to be free-form text then no need to validate it. :+1:

drh-stanford commented:
I'm :+1: on this
